### PR TITLE
ucblogo: update to 6.2.2 from github

### PR DESCRIPTION
**Pull request by srirangav**
#### Description

Per advice from @ryandesign, I have updated the Portfile for ucblogo to use the latest release on github.  Also per advice from @ryandesign, I have removed the patch files that I previously prepared as they appear to no longer be necessary to build this port when using the latest release from github.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.6.5 20G527 arm64
Xcode 13.2.1 13C100

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [X] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/ucblogo/Portfile [hunk 1]
```diff
@@ -1,38 +1,26 @@
...
-revision         0
-categories       lang
-maintainers      nomaintainer
-platforms        darwin
-description      Berkeley Logo
-long_description \
-    Berkeley Logo is a free Logo interpreter.
-homepage         https://people.eecs.berkeley.edu/~bh/logo.html
-
+PortSystem          1.0
```
Comment (ryandesign) [inline lang/ucblogo/Portfile:1]: After your reformatting, the Portfile now conforms to our standard modeline, so you should add it.
```suggestion
# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4

PortSystem          1.0
```
Normally reformatting should be done in a separate commit from functional changes, but since you're changing most of the lines anyway it's probably fine to combine it here.

File: lang/ucblogo/Portfile [hunk 2]
```diff
@@ -1,38 +1,26 @@
...
-	file delete -force "${worksrcpath}/csls/CVS"
-}
+license             GPL-3
 
-configure.args   --cache-file=/dev/null --without-x
+use_autoreconf      yes
+autoreconf.args     --install
 
-build.target
+configure.args      --cache-file=/dev/null --without-x --disable-docs
```
Comment (ryandesign) [inline lang/ucblogo/Portfile:25]: This project uses ncurses, and if MacPorts ncurses is installed, it will use it opportunistically. Better to add the dependency.
```suggestion
depends_lib         port:ncurses

configure.args      --cache-file=/dev/null --without-x --disable-docs
```

**Review by ryandesign — CHANGES_REQUESTED** [review]

File: lang/ucblogo/Portfile [hunk 3]
```diff
@@ -1,38 +1,26 @@
...
-description      Berkeley Logo
-long_description \
-    Berkeley Logo is a free Logo interpreter.
-homepage         https://people.eecs.berkeley.edu/~bh/logo.html
-
+PortSystem          1.0
+PortGroup           github 1.0
+
+name                ucblogo
+github.setup        jrincayc ${name}-code 6.2.2
```
[the remark this answers is not in the document]
Comment (srirangav) [reply lang/ucblogo/Portfile]: Hi @ryandesign, 

I have made this change and updated my commit.

Thanks,

-ranga

File: lang/ucblogo/Portfile [hunk 4]
```diff
@@ -1,38 +1,26 @@
...
-                 patch-wrksp_c.diff
-
-post-patch {
-	file delete -force "${worksrcpath}/csls/CVS"
-}
+license             GPL-3
 
-configure.args   --cache-file=/dev/null --without-x
+use_autoreconf      yes
+autoreconf.args     --install
```
[the remark this answers is not in the document]
Comment (srirangav) [reply lang/ucblogo/Portfile]: Hi @ryandesign, 

I have made this change and updated my commit.

Thanks,

-ranga

File: lang/ucblogo/Portfile [hunk 5]
```diff
@@ -1,38 +1,26 @@
...
+checksums           rmd160  2bb21204cd82fe5fdf3d7d9eb05df465d1ad5f6c \
+                    sha256  7269120e395c6745989e5c12b7023e554b8826926c863da2bf522436f5f5921a \
+                    size    823802
+
+categories          lang
+maintainers         nomaintainer
+description         Berkeley Logo
+long_description    Berkeley Logo is a free Logo interpreter.
+homepage            https://people.eecs.berkeley.edu/~bh/logo.html
 # Note that license has changed to GPL-3 in 6.1
```
[the remark this answers is not in the document]
Comment (srirangav) [reply lang/ucblogo/Portfile]: Hi @ryandesign, 

I have removed this comment and updated my commit.

Thanks,

-ranga

File: lang/ucblogo/Portfile [hunk 6]
```diff
@@ -1,38 +1,26 @@
...
+
+name                ucblogo
+github.setup        jrincayc ${name}-code 6.2.2
+revision            1
+
+github.tarball_from archive
+distname            ${github.project}-version_${github.version}
+checksums           rmd160  2bb21204cd82fe5fdf3d7d9eb05df465d1ad5f6c \
+                    sha256  7269120e395c6745989e5c12b7023e554b8826926c863da2bf522436f5f5921a \
+                    size    823802
```
[the remark this answers is not in the document]
Comment (srirangav) [reply lang/ucblogo/Portfile]: Hi @ryandesign, 

I have made this change and updated my commit.

Thanks,

-ranga

File: lang/ucblogo/Portfile (hunk 1 above)
Comment (srirangav) [reply lang/ucblogo/Portfile:3]: Hi @ryandesign, 

I have added this to the portfile.

Thanks,

-ranga

File: lang/ucblogo/Portfile (hunk 2 above)
Comment (srirangav) [reply lang/ucblogo/Portfile:25]: Hi @ryandesign, 

I have added this and updated my commit.

Thanks,

-ranga

File: lang/ucblogo/Portfile [hunk 7]
```diff
@@ -1,38 +1,26 @@
...
-    Berkeley Logo is a free Logo interpreter.
-homepage         https://people.eecs.berkeley.edu/~bh/logo.html
+PortSystem          1.0
+PortGroup           github 1.0
 
-# Note that license has changed to GPL-3 in 6.1
-license          GPL-3
+name                ucblogo
+github.setup        jrincayc ${name}-code 6.2.2 version_
+revision            1
```
Comment (reneeotten) [inline lang/ucblogo/Portfile:8]: I don't think just increasing the `revision` field here will work. You do not change the `version` number, but do change the `master_sites` which caused the `checksums` to change. Therefore shouldn't this be considered a [stealth update](https://trac.macports.org/wiki/PortfileRecipes#stealth-updates)?

Comment (srirangav) [reply lang/ucblogo/Portfile:8]: Hi @reneeotten,

I am not sure what to do here - I originally updated the Portfile to use the version 6.2.2 tarball from the homepage of this project but @ryandesign correctly pointed out to me that this project had moved to github and that correct version 6.2.2 tarball should be pulled from github.  So I updated the Portfile accordingly.  If you have any suggestions on how to handle this, please let me know.

Thanks,

-ranga

Comment (reneeotten) [reply lang/ucblogo/Portfile:8]: the problem here is that in this commit you do not change the versions but the checksums change; that will cause issues if my understanding is correct (@ryandesign feel free to correct me if I'm wrong!) So please read the section I linked above, that explains how to deal with this.

Comment (srirangav) [reply lang/ucblogo/Portfile:8]: Hi @reneeotten,

As recommended in the section you linked, I have added dist_subdir along with a note about removing dist_subdir when the next release comes out.  Hopefully this is the correct way to fix this issue.

Thanks,

-ranga

CC: @ryandesign
